lib/developer_portal/lib/cms/toolbar.rb

```diff
@@ -39,6 +39,30 @@ def draft?
 
   private
 
+  def validate_and_extract_cms_token
+    signature = params.delete(:signature)
+    return signature if signature.blank?
+
+    expires_at = Time.at(params.delete(:expires_at).to_i).utc
```

Comment:
```suggestion
    expires_at = Time.at(params.delete(:expires_at).to_i)
```

btw no need to protect against bad format, because `#to_i` doesn't raise, only resorts to 0 when it doesn't know what to do.

But it actually self-protects itself from being DoSed by a huge input.
```
8] pry(main)> ("1"*1000000000000000000000000000000000000).to_i
RangeError: bignum too big to convert into `long'
from (pry):8:in `*'
```

So after all we probably need somehting like
```suggestion
    expires_at = Time.at(params.delete(:expires_at).to_i rescue PAST_TIME).utc
```

Reply:
This `PAST_TIME` is from the other PR...

Reply:
The issue is that it is reported to us. Then somebody needs to look into it and basically unnecessary stress. That's all.

`PAST_TIME` is just some time in the past. Not reuse it from the other PR.

Reply:
https://github.com/3scale/porta/pull/3938/commits/eaac83b0288dc1cca8328ccf26d01e760d430e21